test/e2e/apimachinery/flowcontrol.go

```diff
@@ -32,40 +32,52 @@ import (
 	"k8s.io/kubernetes/test/e2e/framework"
 )
 
 const (
-	requestConcurrencyLimitMetricName      = "apiserver_flowcontrol_request_concurrency_limit"
-	requestConcurrencyLimitMetricLabelName = "priority_level"
+	requestConcurrencyLimitMetricName  = "apiserver_flowcontrol_request_concurrency_limit"
+	currentExecutingRequestsMetricName = "apiserver_flowcontrol_current_executing_requests"
+	priorityLevelLabelName             = "priority_level"
+	flowSchemaLabelName                = "flow_schema"
 )
 
 var _ = SIGDescribe("API priority and fairness", func() {
 	f := framework.NewDefaultFramework("apf")
 
-	ginkgo.It("should ensure that requests can be classified by testing flow-schemas/priority-levels", func() {
+	ginkgo.It("should ensure that requests are classified correctly", func() {
 		testingFlowSchemaName := "e2e-testing-flowschema"
 		testingPriorityLevelName := "e2e-testing-prioritylevel"
 		matchingUsername := "noxu"
 		nonMatchingUsername := "foo"
 
-		ginkgo.By("creating a testing prioritylevel")
+		ginkgo.By("creating a testing priority level")
 		createdPriorityLevel, cleanup := createPriorityLevel(f, testingPriorityLevelName, 1)
 		defer cleanup()
 
-		ginkgo.By("creating a testing flowschema")
+		ginkgo.By("creating a testing flow schema")
 		createdFlowSchema, cleanup := createFlowSchema(f, testingFlowSchemaName, 1000, testingPriorityLevelName, []string{matchingUsername})
 		defer cleanup()
 
-		ginkgo.By("checking response headers contain flow-schema/priority-level uid")
-		if !testResponseHeaderMatches(f, matchingUsername, string(createdPriorityLevel.UID), string(createdFlowSchema.UID)) {
-			framework.Failf("matching user doesnt received UID for the testing priority-level and flow-schema")
-		}
-		if testResponseHeaderMatches(f, nonMatchingUsername, string(createdPriorityLevel.UID), string(createdPriorityLevel.UID)) {
-			framework.Failf("non-matching user unexpectedly received UID for the testing priority-level and flow-schema")
-		}
+		ginkgo.By("waiting for testing flow schema and priority level to reach steady state")
+		waitForSteadyState(f, flowSchemaName, priorityLevelName)
+
+		ginkgo.By("retrieving catch-all flow schema and priority level objects")
+		globalDefaultFlowSchema := getFlowSchema(f, "global-default")
+		globalDefaultPriorityLevel := getPriorityLevelConfig(f, globalDefaultFlowSchema.Spec.PriorityLevelConfiguration.Name)
+
+		var response *http.Response
+		ginkgo.By("checking response headers contain flow schema and priority level UID for a matching user")
+		response = makeRequest(f, matchingUsername)
+		verifyResponseHeaderMatch(f, response, string(createdPriorityLevel.UID), string(createdFlowSchema.UID))
+
+		ginkgo.By("checking response headers contain catch-all flow schema and priority level UID for a non-matching user")
+		response = makeRequest(f, nonMatchingUsername)
+		verifyResponseHeaderMatch(f, response, string(globalDefaultPriorityLevel.UID), string(globalDefaultFlowSchema.UID))
```

Comment:
I still do not think this test should insist on particular UIDs in the case of a request that does not match the FlowSchema that this test creates.

Reply:
@MikeSpreitzer this PR builds on top of https://github.com/kubernetes/kubernetes/pull/96966. I have already addressed this feedback in 96966. 

@adtac I think this PR can merge independently of https://github.com/kubernetes/kubernetes/pull/96966. You can copy `getFlowSchema` and `getPriorityLevelConfig` function in this PR. I am no longer using them in https://github.com/kubernetes/kubernetes/pull/96966

Reply:
Done, removed the cherry-picked commit and moved getFlowSchema into this commit